addons/pos_self_order/models/pos_config.py

```diff
@@ -158,7 +161,21 @@ def write(self, vals):
             if vals.get('self_ordering_mode') == 'mobile' and vals.get('self_ordering_pay_after') == 'meal':
                 vals['self_ordering_service_mode'] = 'table'
 
+            for attachment_field in ("self_ordering_image_background_ids", "self_ordering_image_home_ids"):
+                cmds = vals.get(attachment_field, [])
+                for cmd in cmds:
+                    if cmd[0] == Command.CREATE:
+                        cmd[2]["public"] = True
+                    elif cmd[0] == Command.LINK and cmd[1]:
+                        attachment_ids.add(cmd[1])
+                    elif cmd[0] == Command.SET and cmd[2]:
+                        attachment_ids.update(cmd[2])
+
         res = super().write(vals)
+
+        if attachment_ids:
+            self.env['ir.attachment'].browse(attachment_ids).write({"public": True})
```

Comment:
```suggestion
        res = super().write(vals)

        if vals.get('self_ordering_image_background_ids', False):
            self.self_ordering_image_background_ids.filtered(lambda a: not a.public).write({'public': True})

        if vals.get('self_ordering_image_home_ids', False):
            self.self_ordering_image_home_ids.filtered(lambda a: not a.public).write({'public': True})
```

Reply:
Filtered isn't even needed

Reply:
You can also add this in a method like `_ensure_public_attachments` and call it in write & create like that you don't need to specify public in `_prepare_self_order_splash_screen`.